### PR TITLE
Add an upgrade step to fix broken-by-broken-protocol-excerpt journal entries on dossiers

### LINKED ISSUE #4587: UnicodeDecodeError: Beim Darstellen des Journal Tabs in einem Dossier
https://sentry.4teamwork.ch/sentry/onegov-gever/issues/3975/

```
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 92: ordinal not in range(128)
(9 additional frame(s) were not displayed)
...
  File "apps/eggs/zope.i18n-3.7.4-py2.7.egg/zope/i18n/__init__.py", line 126, in translate
    return util.translate(msgid, mapping, context, target_language, default)
  File "apps/eggs/zope.i18n-3.7.4-py2.7.egg/zope/i18n/translationdomain.py", line 78, in translate
    msgid, mapping, target_language, default, context)
  File "apps/eggs/zope.i18n-3.7.4-py2.7.egg/zope/i18n/translationdomain.py", line 140, in _recursive_translate
    text = interpolate(text, mapping)
  File "apps/eggs/zope.i18n-3.7.4-py2.7.egg/zope/i18n/__init__.py", line 164, in interpolate
    return _interp_regex.sub(replace, text)
  File "apps/eggs/zope.i18n-3.7.4-py2.7.egg/zope/i18n/__init__.py", line 159, in replace
    return unicode(mapping.get(param1 or param2, whole))

UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 92: ordinal not in range(128)
```

### PR DESCRIPTION
**Pull request by Rotonen**
Seems our saga with the protocol excerpt titles was not yet done.

@phgross @deiferni a candidate for a backport?

Closes #4587

### CODE REVIEW

File: opengever/core/upgrades/20180904184647_fix_dossier_journal_entries_of_broken_titled_meeting_excerpts/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        excerpts = self.execute(
+            generateddocuments_table.select()
+            .where(generateddocuments_table.c.admin_unit_id == admin_unit_id)
+            .where(generateddocuments_table.c.generated_document_type == 'generated_excerpt')
+            ).fetchall()
+        if not excerpts:
+            return
+        journal_fixer = JournalMessageFixer(keys_to_fix=['title'])
+        intids = getUtility(IIntIds)
+        dossiers = set(intids.queryObject(excerpt.int_id).get_parent_dossier() for excerpt in excerpts)
```
Comment (deiferni) [inline opengever/core/upgrades/20180904184647_fix_dossier_journal_entries_of_broken_titled_meeting_excerpts/upgrade.py]: there may be rare cases where the `int_id` points to `None`. Even though this is invalid state i would be careful here and handle `queryObject` returning `None`.

File: opengever/core/upgrades/20180904184647_fix_dossier_journal_entries_of_broken_titled_meeting_excerpts/upgrade.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    )
+
+
+class FixDossierJournalEntriesOfBrokenTitledMeetingExcerpts(SQLUpgradeStep):
+    """Fix dossier journal entries of broken-titled meeting excerpts."""
+
+    def migrate(self):
+        has_meeting_feature = api.portal.get_registry_record('opengever.meeting.interfaces.IMeetingSettings.is_feature_enabled')  # noqa
+        if not has_meeting_feature:
+            return
```
Comment (deiferni) [inline opengever/core/upgrades/20180904184647_fix_dossier_journal_entries_of_broken_titled_meeting_excerpts/upgrade.py]: IMO adding an extra newline in appropriate places would improve readability, for example in this method after each `return`.

**Comment by Rotonen** [comment]
Ready for rereview.

**Comment by Rotonen** [comment]
@phgross so now that we still do have an another bugfix release, do we backport this after all?

**Comment by phgross** [comment]
@Rotonen I think this bug is not critical and there is no bugfix-release planned for our SPV customers, so I would say no backport yet.

**Pull request merged**